src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IfError.cs

```diff
@@ -96,75 +96,60 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
+                    fArgsValid = false;
                 }
-
-                // In an IfError expression, not all expressions can be returned to the caller:
-                // - If there is an even number of arguments, only the fallbacks or the last
-                //   value (the next-to-last argument) can be returned:
-                //   IfError(v1, f1, v2, f2, v3, f3) --> possible values to be returned are f1, f2, f3 or v3
-                // - If there is an odd number of arguments, only the fallbacks or the last
-                //   value (the last argument) can be returned:
-                //   IfError(v1, f1, v2, f2, v3) --> possible values to be returned are f1, f2 or v3
-                var typeCanBeReturned = (i % 2) == 1 || i == ((count % 2) == 0 ? (count - 2) : (count - 1));
-
-                if (typeCanBeReturned)
+                else if (!type.IsError)
                 {
-                    // Let's check if it matches the other types that can be returned
-                    var typeSuper = DType.Supertype(type, typeArg);
-
-                    if (!typeSuper.IsError)
+                    // Types don't resolve normally, coercion needed
+                    if (typeArg.CoercesTo(type))
                     {
-                        type = typeSuper;
+                        CollectionUtils.Add(ref nodeToCoercedTypeMap, nodeArg, type);
                     }
-                    else if (type.Kind == DKind.Unknown)
+                    else
                     {
-                        // One of the args is also of unknown type, so we can't resolve the type of IfError
-                        type = typeSuper;
-                        fArgsValid = false;
-                    }
-                    else if (!type.IsError)
-                    {
-                        // Types don't resolve normally, coercion needed
-                        if (typeArg.CoercesTo(type))
-                        {
-                            CollectionUtils.Add(ref nodeToCoercedTypeMap, nodeArg, type);
-                        }
-                        else if (!isBehavior || !IsArgTypeInconsequential(nodeArg))
-                        {
-                            errors.EnsureError(
-                                DocumentErrorSeverity.Severe,
-                                nodeArg,
-                                TexlStrings.ErrBadType_ExpectedType_ProvidedType,
-                                type.GetKindString(),
-                                typeArg.GetKindString());
-                            fArgsValid = false;
-                        }
-                    }
-                    else if (typeArg.Kind != DKind.Unknown)
-                    {
-                        type = typeArg;
-                        fArgsValid = false;
+                        // If the types are incompatible, the result type is void.
+                        type = DType.Void;
+                        break;
```

Comment:
>break [](http://example.com/codeflow?start=24&length=5)

Should we continue (instead of breaking) to see if there are any arguments with error types?

Reply:
you are correct, removed the `break;`